BlazarUI/app/scripts/components/module/BuildButton.jsx

```diff
@@ -0,0 +1,52 @@
+import React, {Component, PropTypes} from 'react';
+import {Button} from 'react-bootstrap';
+import Icon from '../shared/Icon.jsx';
+import $ from 'jquery';
+
+class BuildButton extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {loading: false, textStatus: ''};
+  }
+
+  build() {
+    this.setState({
+      loading: true
+    });
+    $.ajax({
+      url: `${config.apiRoot}/build/module/${this.props.moduleId}`,
+      type: 'POST'
+    }).then(() => {
+      this.setState({
+        loading: false,
+        textStatus: ''
+      });
+    },
+    (data, textStatus, jqXHR) => {
+      this.setState({
+        loading: false,
+        textStatus: jqXHR
+      });
+    });
+  }
+
+  render() {
+    if (this.state.loading) {
+      return <Button bsStyle="primary" disabled><Icon type="fa" name="circle-o-notch" classNames="fa-spin" /> Build Now</Button>;
```

Comment:
I added a new prop to the [Icon component](https://github.com/HubSpot/Blazar/blob/master/BlazarUI/app/scripts/components/shared/Icon.jsx#L11) called `for` with a mapping in the [constants file](https://github.com/HubSpot/Blazar/blob/master/BlazarUI/app/scripts/components/constants.js#L24) so we don't need to worry about the exact names of the fa and octicon icons.  I still need to go back to update old Icons, but moving forward let's use this.